source/initial-dns-seedlist-discovery/initial-dns-seedlist-discovery.rst

```diff
@@ -61,35 +64,40 @@ format is::
 specification following the ``Host Information``. This includes the ``Auth
 database`` and ``Connection Options``.
 
-Seedlist Discovery
-------------------
 
-In this preprocessing step, the driver will query the DNS server for SRV
-records on ``{hostname}.{domainname}``, prefixed with the SRV service name
-and protocol. The SRV service name is provided in the ``srvServiceName`` URI option and
-defaults to ``mongodb``. The protocol is always ``tcp``. After prefixing, the URI
-should look like: ``_{srvServiceName}._tcp.{hostname}.{domainname}``. This DNS query
-is expected to respond with one or more SRV records. The driver MUST add all the host
-names and port numbers that were returned as part of the DNS SRV query result to the
-seedlist.
+MongoClient Configuration
+-------------------------
 
-The priority and weight fields in returned SRV records MUST be ignored.
+srvMaxHosts
+~~~~~~~~~~~
 
-If ``mongodb+srv`` is used, a driver MUST implicitly also enable TLS. Clients
-can turn this off by passing ``ssl=false`` in either the Connection String,
-or options passed in as parameters in code to the MongoClient constructor (or
-equivalent API for each driver), but not through a TXT record (discussed in
-the next section).
+This option may be used to limit the number of SRV records used to populate the
+seedlist during initial discovery. It is also used to limit the number of
+additional hosts that may be added during
+`SRV polling <../polling-srv-records-for-mongos-discovery/polling-srv-records-for-mongos-discovery.rst>`_.
+This option requires a non-negative integer and defaults to zero (i.e. no
+limit). This option MUST only be configurable at the level of a ``MongoClient``.
 
-A driver MUST verify that in addition to the ``{hostname}``, the
-``{domainname}`` consists of at least two parts: the domain name, and a TLD.
-Drivers MUST raise an error and MUST NOT contact the DNS server to obtain SRV
-(or TXT records) if the full URI does not consists of at least three parts.
+This option is primarily used to limit the number of mongos connections that may
+be created for sharded topologies.
```

Comment:
@benjirewis pointed out that [`srvServiceName` is prohibited for non-SRV schemes](https://github.com/mongodb/specifications/blob/master/source/uri-options/uri-options.rst#non-srv-uri-with-srvservicename-uri-option) (tested by [srv-service-name-option.yml](https://github.com/mongodb/specifications/blob/master/source/uri-options/tests/srv-service-name-option.yml)). That makes sense, so I'll revise this PR to make the same prohibition for `srvMaxHosts` and add a corresponding URI test.

I do think it'd be worth repeating that in the SRV spec itself (not just URI options), so I'll amend both of these sections as well.

Reply:
I'd get rid of the word "primarily". There isn't, currently, any other use case.